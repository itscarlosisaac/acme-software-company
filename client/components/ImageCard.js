import React, { Component } from 'react'
import EventEmitter from '../events/emitter';

export default class ImageCard extends Component {

  handleClick = () => {
    // EventEmitter.emit('thumbClick', this.props)
    this.props.handleThumbClick(this.props.id)Â 
  }

  render() {
    return (
      <div className="image__card" onClick={this.handleClick}>
        <div className="image__card--thumb">
          <img src={this.props.image} alt={this.props.title} />
        </div>
        <h3 className="image__card--title">{this.props.title}</h3>
      </div>
    )
  }
}
