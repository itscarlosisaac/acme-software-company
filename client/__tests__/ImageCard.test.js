import React from 'react';
import ImageCard from '../components/ImageCard';

import { configure, shallow } from 'enzyme';
import Adapter from  'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';


configure({adapter: new Adapter() });

const imageURL = 'https://media0.giphy.com/media/3o7TKqLfSYZwuyJfCo/giphy_s.gif'
const title = 'My image';

describe('<LoadMoreButton>', () => {
  it('Should render component properly', () => {
    const wrapper = shallow(<ImageCard image={imageURL} title={title} handleThumbClick={()=> {}} />)
    expect(wrapper.find('img').props().src).toBe(imageURL);
    expect(wrapper.find('img').props().alt).toBe(title);
    expect(wrapper.find('h3').text()).toBe(title);
    const f = typeof wrapper.instance().props.handleThumbClick;
    expect(f).toBe('function')
  });

  it(`Should call the handleThumbClick on click`, () => {
    const mockCB = jest.fn();
    const wrapper = shallow(<ImageCard image={imageURL} title={title} handleThumbClick={mockCB} />)
    wrapper.simulate('click')
    expect(mockCB.mock.calls.length).toEqual(1)
  });

  it(`Matches the snapshot`, () => {
    const tree = shallow(<ImageCard image={imageURL} title={title} handleThumbClick={()=> {}} />)
    expect(toJson(tree)).toMatchSnapshot();
  });
});