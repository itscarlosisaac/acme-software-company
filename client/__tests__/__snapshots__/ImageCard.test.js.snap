// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadMoreButton> Matches the snapshot 1`] = `
<div
  className="image__card"
  onClick={[Function]}
>
  <div
    className="image__card--thumb"
  >
    <img
      alt="My image"
      src="https://media0.giphy.com/media/3o7TKqLfSYZwuyJfCo/giphy_s.gif"
    />
  </div>
  <h3
    className="image__card--title"
  >
    My image
  </h3>
</div>
`;
